### PR TITLE
Make GreenNode.CreateList static

### PR DESCRIPTION
**Pull request by HurricanKai**
As discussed with @CyrusNajmabadi on [discord](https://discord.com/channels/143867839282020352/598678594750775301/765314698261692428) this PR makes GreenNode.CreateList static and adjusts all usages.

This done to simplify the method and remove the virtual call which should help performance somewhat, but I couldn't get perf numbers that could directly confirm this. 

I'm not sure whether members should also be reordered. I left them as is for now.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=48536) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=48536) before we can accept your contribution.<br/><br/>:x: HurricanKai [sign now](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=48536)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/roslyn?pullRequest=48536) it.</sub>

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Compilers/Core/Portable/Syntax/SyntaxList`1.cs [hunk 1]
```diff
@@ -327,7 +327,7 @@ private static SyntaxList<TNode> CreateList(GreenNode creator, List<TNode> items
                 return default(SyntaxList<TNode>);
             }
 
-            var newGreen = creator.CreateList(items.Select(n => n.Green));
+            var newGreen = GreenNode.CreateList(items.Select(n => n.Green));
```
Comment (333fred) [inline src/Compilers/Core/Portable/Syntax/SyntaxList`1.cs:330]: It looks like `cretor` is also unused now. We should remove these unecessary parameters all the way up the tree.

**Review by 333fred — CHANGES_REQUESTED** [review]
Done review pass (commit 1)

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler can we get a second pair of eyes on this nice community PR? :)

**Review by 333fred — APPROVED** [review]
LGTM (commit 4). @dotnet/roslyn-compiler for a second review. There are no perf numbers in this PR, but Kai is planning on submitting a second PR that actually refactors the implementation of `GreenNode.CreateList` that removes a bunch of double-allocations that should have a big impact, and this first change will simplify that.

**Comment by CyrusNajmabadi** [comment]
> There are no perf numbers in this PR,

True.  Though changing these virtual methods (and extra parameters and whatnot) to just static calls, should be pure benefit.  At worse, we should have no perf change.  A best, potentially some uptick :)

**Review by cston — APPROVED** [review]

**Comment by 333fred** [comment]
Thanks @HurricanKai!

**Pull request merged**